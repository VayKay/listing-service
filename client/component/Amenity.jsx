import React from 'react';
import styles from './style/Amenity.css';

class Amenity extends React.Component{
    constructor(props){
        super(props)
    }
    showamenlist(){
        this.props.showamenitylist()
    }
   
    render(){ 
        if(this.props.amenity){
            var basicitems = this.props.amenity.Basic
        }
        if(basicitems){
            var amenitems = Object.keys(basicitems)
            var amenlength = 0;
            var allCategoryAmen = Object.keys(this.props.amenity);
            for(var i=0; i<allCategoryAmen.length; i++){
                amenlength += Object.keys(this.props.amenity[allCategoryAmen[i]]).length
            }
        }
        var amenIcon = {
            parking: 'm12 0c-6.63 0-12 5.37-12 12s5.37 12 12 12 12-5.37 12-12-5.37-12-12-12zm0 23c-6.07 0-11-4.92-11-11s4.93-11 11-11 11 4.93 11 11-4.93 11-11 11zm .5-17h-4.5v11.5a.5.5 0 0 0 1 0v-4.5h3.5c1.93 0 3.5-1.57 3.5-3.5s-1.57-3.5-3.5-3.5zm0 6h-3.5v-5h3.5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5z',
            wifi: 'm12 15a3 3 0 1 0 0 6 3 3 0 0 0 0-6zm0 5a2 2 0 1 1 0-4 2 2 0 0 1 0 4zm5.92-5.78a.5.5 0 1 1 -.84.55c-1.19-1.81-3.07-2.77-5.08-2.77s-3.89.96-5.08 2.78a.5.5 0 0 1 -.84-.55c1.38-2.1 3.58-3.23 5.92-3.23s4.54 1.13 5.92 3.23zm2.98-3.03a.5.5 0 1 1 -.79.61c-1.66-2.14-5.22-3.8-8.11-3.8-2.83 0-6.26 1.62-8.12 3.82a.5.5 0 0 1 -.76-.65c2.05-2.42 5.75-4.17 8.88-4.17 3.19 0 7.05 1.8 8.9 4.19zm2.95-2.33a.5.5 0 0 1 -.71-.02c-2.94-3.07-6.71-4.84-11.14-4.84s-8.2 1.77-11.14 4.85a.5.5 0 0 1 -.72-.69c3.12-3.27 7.14-5.16 11.86-5.16s8.74 1.89 11.86 5.16a.5.5 0 0 1 -.02.71z',
            tv: 'm21.5 6h-9.07l3.49-5.22a.5.5 0 1 0 -.83-.55l-3.86 5.77h-2.46l-3.85-5.78a.5.5 0 0 0 -.83.55l3.48 5.23h-5.07a2.5 2.5 0 0 0 -2.5 2.5v13c0 1.39 1.12 2.5 2.5 2.5h19c1.38 0 2.5-1.12 2.5-2.5v-13c0-1.38-1.12-2.5-2.5-2.5zm1.5 15.5c0 .83-.68 1.5-1.5 1.5h-19c-.83 0-1.5-.67-1.5-1.5v-13c0-.83.68-1.5 1.5-1.5h19c .83 0 1.5.67 1.5 1.5zm-6.49-12.5h-11.02a2.49 2.49 0 0 0 -2.49 2.49v7.02a2.5 2.5 0 0 0 2.49 2.49h11.02a2.49 2.49 0 0 0 2.49-2.49v-7.02a2.5 2.5 0 0 0 -2.49-2.49zm1.49 9.51c0 .82-.67 1.49-1.49 1.49h-11.02c-.82 0-1.49-.67-1.49-1.49v-7.02c0-.82.67-1.49 1.49-1.49h11.02c.82 0 1.49.67 1.49 1.49zm4-8.51a1 1 0 1 1 -2 0 1 1 0 0 1 2 0zm0 3a1 1 0 1 1 -2 0 1 1 0 0 1 2 0z',
            kitchen: 'm10.5 0a .5.5 0 0 0 -.5.5v7a .5.5 0 0 1 -.49.5h-1.51v-7.5a.5.5 0 1 0 -1 0v7.5h-1.51a.5.5 0 0 1 -.49-.5v-7a .5.5 0 1 0 -1 0v7c0 .83.67 1.5 1.49 1.5h1.51v5c0 .03.01.06.02.09a1.49 1.49 0 0 0 -1.02 1.41v7c0 .83.67 1.5 1.5 1.5s1.5-.67 1.5-1.5v-7c0-.66-.43-1.21-1.02-1.41.01-.03.02-.06.02-.09v-5h1.51a1.5 1.5 0 0 0 1.49-1.5v-7a .5.5 0 0 0 -.5-.5zm-2.5 15.5v7a .5.5 0 0 1 -.5.5.5.5 0 0 1 -.5-.5v-7c0-.28.22-.5.5-.5s.5.22.5.5zm11.5-15.5h-2c-1.4 0-2.5 1.07-2.5 2.5v7c0 1.43 1.1 2.5 2.5 2.5h1.5v2.09a1.49 1.49 0 0 0 -.5-.09c-.83 0-1.5.67-1.5 1.5v7c0 .83.67 1.5 1.5 1.5s1.5-.67 1.5-1.5v-22.5zm-2 11c-.86 0-1.5-.63-1.5-1.5v-7c0-.87.65-1.5 1.5-1.5h1.5v10zm1.5 11.5a.5.5 0 0 1 -.5.5.5.5 0 0 1 -.5-.5v-7c0-.28.22-.5.5-.5s.5.22.5.5z',
            iron: 'm23.99 18.38-.5-2a .5.5 0 0 0 -.49-.38h-22a .5.5 0 0 0 -.49.38l-.5 2a .5.5 0 0 0 .49.62h23a .5.5 0 0 0 .49-.62zm-1.13-.38h-21.72l.25-1h21.22zm-21.36-3h21a .5.5 0 0 0 .5-.53c-.21-3.22-1.22-5.47-3-5.47a4911.8 4911.8 0 0 0 -8.8 0h-1.71c-.2 0-.26-.08-.19-.27a9.59 9.59 0 0 1 .17-.48c.13-.34.27-.68.43-1 .41-.79.82-1.25 1.1-1.25h10.5c.87 0 1.43-.7 1.4-1.52s-.64-1.48-1.55-1.48h-11.35c-3.84 0-7.29 4.4-8.99 11.38a.5.5 0 0 0 .49.62zm8.5-11h11.35c.35 0 .55.21.56.52.01.29-.14.48-.4.48h-10.51c-.8 0-1.42.68-1.99 1.8a10.74 10.74 0 0 0 -.65 1.61c-.31.82.23 1.59 1.13 1.59h1.71a33801.74 33801.74 0 0 1 8.8 0c .94 0 1.71 1.58 1.95 4h-19.8c1.65-6.21 4.7-10 7.85-10zm5 8a1 1 0 1 1 2 0 1 1 0 0 1 -2 0zm3 0a1 1 0 1 1 2 0 1 1 0 0 1 -2 0z',
            fireplace: 'm23.5 23h-1.5v-19a2 2 0 0 0 -.01-4h-19.98a2 2 0 0 0 -.01 4v19h-1.5a.5.5 0 1 0 0 1h23a .5.5 0 0 0 0-1zm-19.5 0h-1v-19h1zm15-19v14h-1.17c.33-1.58.04-3.44-.59-4.72-.24-.5-.49-.93-.92-1.58 0 0-.26-.39-.33-.5-.92-1.43-1.25-2.32-1.16-3.66a.5.5 0 0 0 -.86-.38c-.17.17-.92.88-1.03 1-.8.79-1.29 1.44-1.54 2.19a3.84 3.84 0 0 0 -.23 1.12c-.03.49.01.95.1 1.64l.02.14c-.61-.35-.95-.93-1.14-1.76-.08-.34-.1-.48-.19-1.25a14.33 14.33 0 0 0 -.04-.3.5.5 0 0 0 -.79-.33c-1.8 1.31-3.19 4.37-3.17 7.04 0 .49.07.94.16 1.37h-1.12v-14.02zm-6.72 8.96c-.09-.63-.12-1.03-.09-1.44a2.88 2.88 0 0 1 .17-.85c.2-.58.61-1.13 1.3-1.82l.22-.21c.13 1.02.54 1.91 1.29 3.08l.33.51c.41.62.64 1.02.86 1.47.58 1.18.84 3.01.44 4.4-.52 1.82-2.05 2.89-4.89 2.89-3.31 0-4.89-1.66-4.91-4.38-.02-2 .89-4.28 2.09-5.6.04.28.07.44.13.68.33 1.44 1.1 2.4 2.58 2.78a.5.5 0 0 0 .62-.49c0-.24 0-.26-.11-1.04zm6.72 10.04h-14v-4h1.45c.8 1.87 2.62 3 5.44 3 2.96 0 4.82-1.11 5.63-3h1.48zm2 0h-1v-19h1zm0-20h-18.99a1 1 0 1 1 0-2h19.98a1 1 0 1 1 0 2z',
            dryer: 'm5 3.5a1.5 1.5 0 1 0 -3 0 1.5 1.5 0 0 0 3 0zm-1.5.5a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm4 1a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3zm0-2a .5.5 0 1 1 0 1 .5.5 0 0 1 0-1zm4.5 3c-4.41 0-8 3.59-8 8s3.59 8 8 8 8-3.59 8-8-3.59-8-8-8zm0 15c-3.86 0-7-3.14-7-7s3.14-7 7-7 7 3.14 7 7-3.14 7-7 7zm9.5-21h-19a2.51 2.51 0 0 0 -2.5 2.5v19c0 1.38 1.12 2.5 2.5 2.5h19c1.38 0 2.5-1.12 2.5-2.5v-19c0-1.38-1.12-2.5-2.5-2.5zm1.5 21.5c0 .83-.68 1.5-1.5 1.5h-19c-.83 0-1.5-.68-1.5-1.5v-19c0-.83.68-1.5 1.5-1.5h19c .83 0 1.5.68 1.5 1.5zm-6.17-6.12a.5.5 0 0 1 .04.71 2.49 2.49 0 0 1 -.63.48c-1.15.65-2.67.65-4.49-.38-1.52-.85-2.66-.85-3.51-.38a1.57 1.57 0 0 0 -.37.27.5.5 0 1 1 -.75-.66 2.49 2.49 0 0 1 .63-.48c1.15-.65 2.67-.65 4.49.38 1.52.85 2.66.85 3.51.38a1.57 1.57 0 0 0 .37-.27.5.5 0 0 1 .71-.04zm0-3.5a.5.5 0 0 1 .04.71 2.49 2.49 0 0 1 -.63.48c-1.15.65-2.67.65-4.49-.38-1.52-.85-2.66-.85-3.51-.38a1.57 1.57 0 0 0 -.37.27.5.5 0 1 1 -.75-.66 2.49 2.49 0 0 1 .63-.48c1.15-.65 2.67-.65 4.49.38 1.52.85 2.66.85 3.51.38a1.57 1.57 0 0 0 .37-.27.5.5 0 0 1 .71-.04z',
            washer: 'm5 3.5a1.5 1.5 0 1 0 -3 0 1.5 1.5 0 0 0 3 0zm-1.5.5a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm4 1a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3zm0-2a .5.5 0 1 1 0 1 .5.5 0 0 1 0-1zm4.5 3c-4.41 0-8 3.59-8 8s3.59 8 8 8 8-3.59 8-8-3.59-8-8-8zm0 15c-3.77 0-6.84-2.99-6.99-6.73.35-.23.84-.57.85-.57 1.64-1.05 3.31-1.06 5.85.71 2.88 2.01 4.99 1.99 6.96.73l.25-.17c-.47 3.4-3.39 6.03-6.92 6.03zm6.13-6.7c-1.64 1.05-3.31 1.06-5.85-.71-2.88-2.01-4.99-1.99-6.96-.73l-.25.17c.48-3.4 3.4-6.03 6.93-6.03 3.77 0 6.84 2.99 6.99 6.73-.35.23-.84.57-.85.57zm3.37-14.3h-19a2.51 2.51 0 0 0 -2.5 2.5v19c0 1.38 1.12 2.5 2.5 2.5h19c1.38 0 2.5-1.12 2.5-2.5v-19c0-1.37-1.12-2.5-2.5-2.5zm1.5 21.5c0 .83-.68 1.5-1.5 1.5h-19c-.83 0-1.5-.68-1.5-1.5v-19c0-.82.68-1.5 1.5-1.5h19c .83 0 1.5.68 1.5 1.5z',
            hangers: 'm23.75 16.07-11.75-6.9v-1.23c1.08-.27 2-1.34 2-2.44a2.5 2.5 0 1 0 -5 0 .5.5 0 0 0 1 0 1.5 1.5 0 1 1 3 0c0 .72-.78 1.5-1.5 1.5a.5.5 0 0 0 -.5.5v1.77l-10.77 6.81a.48.48 0 0 0 -.23.41v.01c0 .26.19.51.5.51h23c .32 0 .51-.26.5-.52v-.01a.48.48 0 0 0 -.25-.41zm-21.52-.07 9.36-5.92 10.07 5.92z',
        }   
        var amenkeys = Object.keys(amenIcon);
        
        return(
            <div className={styles.amenityarea} >                
                <div className={styles.amenhead}>Amenities</div>
                <div className={styles.amenarea}>
                {amenitems &&
                    <div className={styles.amenarealeft}>             
                        <div className={styles.amenitem} key={amenitems[0]}>
                            {amenkeys.map(icon=> {
                                if(amenitems[0].toLowerCase().includes(icon)){
                                    return (<svg viewBox={'0 0 24 24'} role={'presentation'} focusable={'false'}  key={amenitems[0]}>
                                        <path d={amenIcon[icon]} ></path>
                                    </svg>)
                                } 
                            })}
                            {amenitems[0]}
                        </div>
                        <div className={styles.amenitem}  key={amenitems[1]}>
                            {amenkeys.map(icon=> {
                                if(amenitems[1].toLowerCase().includes(icon)){
                                    return (<svg viewBox={'0 0 24 24'} role={'presentation'} focusable={'false'}  key={amenitems[1]}>
                                        <path d={amenIcon[icon]} ></path>
                                    </svg>)
                                } 
                            })}
                            {amenitems[1]}
                        </div>
                    </div>
                }
                 {amenitems &&
                    <div className={styles.amenarearight}>             
                        <div className={styles.amenitem} key={amenitems[2]}>
                            {amenkeys.map(icon=> {
                                if(amenitems[2].toLowerCase().includes(icon)){
                                    return (<svg viewBox={'0 0 24 24'} role={'presentation'} focusable={'false'}  key={amenitems[2]}>
                                        <path d={amenIcon[icon]} ></path>
                                    </svg>)
                                } 
                            })}
                            {amenitems[2]}
                        </div>
                        <div className={styles.amenitem} key={amenitems[3]}>
                            {amenkeys.map(icon=> {
                                if(amenitems[3].toLowerCase().includes(icon)){
                                    return (<svg viewBox={'0 0 24 24'} role={'presentation'} focusable={'false'} key={amenitems[3]} >
                                        <path d={amenIcon[icon]} ></path>
                                    </svg>)
                                } 
                            })}
                            {amenitems[3]}
                        </div>
                    </div>
                }      
                </div>
                <div className={styles.moreamen}>
                    <button className={styles.btnMoreamen} onClick={e=>this.showamenlist()} >
                        Show all {amenlength} amenities
                    </button>
                </div>
            </div>
        )
    }
}
export default Amenity;