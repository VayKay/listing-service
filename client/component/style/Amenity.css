.amenhead{
    margin: 0px;
    word-wrap: break-word ;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif;
    font-size: 16px ;
    font-weight: 600 ;
    line-height: 1.375em ;
    color: #484848;
    margin-bottom: 16px;
}
.moreamen{
    margin-top: 8px;
}
.btnMoreamen{
    color: #008489;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif;
    text-decoration: none;
    word-wrap: break-word ;
    font-size: 16px ;
    font-weight: 600 ;
    line-height: 1.375em;
    border: none;
    padding: 0;
}
.btnMoreamen:hover{
    text-decoration: underline
}
.amenarea{
    display: flex;
    flex-direction: row;
}
.amenarealeft, .amenarearight{
    padding-left: 8px ;
    padding-right: 8px ;
    min-height: 1px ;
    position: relative ;
    width: 50% ;
}

.amenitem{
    margin: 0px ;
    padding-left: 8px ;
    padding-right: 8px ;
    word-wrap: break-word ;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif ;
    font-size: 16px ;
    font-weight: 200 ;
    line-height: 1.375em ;
    color: #484848;
    margin-bottom: 16px; 
    display: flex;
    align-items: center;
}
.amenitem svg{
    height: 19px;
    width: 19px;
    fill: currentcolor;
    padding: 1px;
    margin-right: 5px;
}