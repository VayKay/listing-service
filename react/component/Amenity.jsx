import React from 'react';
class Amenity extends React.Component{
    constructor(props){
        super(props)
    }
    showamenlist(){
        this.props.showamenitylist()
    }
    render(){ 
        // console.log('props in Amenity',this.props)  
        if(this.props.amenity){
            var basicitems = this.props.amenity.Basic
        }
        if(basicitems){
            var amenitems = Object.keys(basicitems)
            var amenlength = amenitems.length
        }
        // console.log('items',amenitems);
        // console.log('length',amenlength)
        return(
            <div id='amenityarea' >                
                <div id='amenhead'>Amenities</div>
                <div id='amenarea'>
                {amenitems &&
                    <div id='amenarealeft'>             
                        <div className='amenitem'>{amenitems[0]}</div>
                        <div className='amenitem'>{amenitems[1]}</div>
                    </div>
                }
                 {amenitems &&
                    <div id='amenarearight'>             
                        <div className='amenitem'>{amenitems[2]}</div>
                        <div className='amenitem'>{amenitems[3]}</div>
                    </div>
                }      
                </div>
                <div id='moreamen'>
                    <button id='btn-moreamen' onClick={e=>this.showamenlist()}>
                        Show all {amenlength} amenities
                    </button>
                </div>
            </div>
        )
    }
}
export default Amenity;