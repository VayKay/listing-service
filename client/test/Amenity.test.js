import React from 'react';
import Amenity from '../component/Amenity.jsx'
import { shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
require('babel-core/register');
require('babel-polyfill');

configure({ adapter: new Adapter() });

describe('Amenity test suite', () => {
    it('render App' , ()=>{
        const amenityWrapper = shallow(<Amenity />);
        amenityWrapper.setProps({
            amenity: {
                Basic:{
                    'Air conditioning': null,
                    'Cable TV': null,
                    'Essentials': null,
                    'Hair dryer': null
                }
            }
        });
        // console.log(amenityWrapper.debug())
        expect(amenityWrapper.exists()).toBe(true);
    })
    it('snapshots', async()=>{
        const amenityWrapper = shallow(<Amenity />);
        amenityWrapper.setProps({
            amenity: {
                Basic:{
                    'Air conditioning': null,
                    'Cable TV': null,
                    'Essentials': null,
                    'Hair dryer': null
                }
            }
        });
        expect(toJson(amenityWrapper)).toMatchSnapshot()
    })
})