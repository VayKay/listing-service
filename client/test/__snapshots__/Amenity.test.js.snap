// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Amenity test suite snapshots 1`] = `
<div
  id="amenityarea"
>
  <div
    id="amenhead"
  >
    Amenities
  </div>
  <div
    id="amenarea"
  >
    <div
      id="amenarealeft"
    >
      <div
        className="amenitem"
      >
        Air conditioning
      </div>
      <div
        className="amenitem"
      >
        Cable TV
      </div>
    </div>
    <div
      id="amenarearight"
    >
      <div
        className="amenitem"
      >
        Essentials
      </div>
      <div
        className="amenitem"
      >
        Hair dryer
      </div>
    </div>
  </div>
  <div
    id="moreamen"
  >
    <button
      id="btn-moreamen"
      onClick={[Function]}
    >
      Show all 
      4
       amenities
    </button>
  </div>
</div>
`;
